src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindOutputPresenter.java

```diff
@@ -143,13 +191,192 @@ public void onFindOperationEnded(
          {
             if (event.getHandle() == currentFindHandle_)
             {
-               currentFindHandle_ = null;
+               if (!view_.getReplaceMode())
+                  currentFindHandle_ = null;
                view_.setStopSearchButtonVisible(false);
                view_.showSearchCompleted();
             }
          }
       });
 
+      events_.addHandler(PreviewReplaceEvent.TYPE, new PreviewReplaceEvent.Handler()
+      {
+         @Override
+         public void onPreviewReplace(PreviewReplaceEvent event)
+         {
+            view_.setRegexPreviewMode(true);
```

Comment:
Yes it should be. We only send the Preview Replace RPC for regexes so the backend will determine the outcome exactly as it will when performing the replace. For non regexes, the frontend can handle the preview.